src/ape/contracts/base.py

```diff
@@ -1300,6 +1300,13 @@ def __call__(self, *args, **kwargs) -> TransactionAPI:
     def deploy(self, *args, publish: bool = False, **kwargs) -> ContractInstance:
         txn = self(*args, **kwargs)
         private = kwargs.get("private", False)
+        has_value = kwargs.get("value")
+
+        if has_value:
+            has_non_payable_constructor = self.contract_type.constructor.is_payable
+            # print(has_non_payable_constructor)
+            if not has_non_payable_constructor:
+                raise ContractError("Sending funds to a non-payable constructor!!")
```

Comment:
```suggestion

        if kwargs.get("value") and not self.contract_type.constructor.is_payable
            raise ContractError("Sending funds to a non-payable constructor.")
```

Reply:
Also can probably make a new exception in `ape/exceptions.py` for this instead, since it's pre-contract logic